backend/nomad/jobs/restart.go

```diff
@@ -0,0 +1,81 @@
+package jobs
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/jippi/hashi-ui/backend/structs"
+	//"encoding/json"
+	//"log"
+	"encoding/json"
+	"log"
+)
+
+const (
+	Restart = "NOMAD_RESTART_JOB"
+)
+
+type restart struct {
+	action structs.Action
+	client *api.Client
+}
+
+func NewRestart(action structs.Action, client *api.Client) *restart {
+	return &restart{
+		action: action,
+		client: client,
+	}
+}
+
+func (w *restart) Do() (*structs.Response, error) {
+	origJob, _, err := w.client.Jobs().Info(w.action.Payload.(string), nil)
+	if err != nil {
+		return structs.NewErrorResponse(err)
+	}
+
+	// stop
+	_, _, err = w.client.Jobs().Deregister(w.action.Payload.(string), false, nil)
+	if err != nil {
+		return structs.NewErrorResponse(err)
+	}
+
+	// start
+	// do some json-dancing to modify the structure to make sure the job is in to-be-started-state
+	jobJson, err := json.Marshal(origJob)
```

Comment:
you can just do do `origJob.Stop = false` ?

Reply:
I tried to do job.Stop = false, but it doesn't allow that (It's a pointer to a bool?)
I do the explicit stop=false in case the job was stopped before and the user just wants to start it.

Reply:
origJob.Stop=false:
https://snag.gy/FT6Yrb.jpg

origJob.Stop=&false:
https://snag.gy/RLF98p.jpg
(which makes sense, I guess)

Reply:
use this to make the pool

```go
// boolToPtr returns the pointer to a bool
func boolToPtr(b bool) *bool {
	return &b
}
```

and then

```go
job.Stop = boolToPtr(false)
```

Reply:
thx, will try!

Reply:
Works!